static/app/views/replays/replayTable/index.tsx

```diff
@@ -0,0 +1,136 @@
+import {Fragment} from 'react';
+import styled from '@emotion/styled';
+
+import Alert from 'sentry/components/alert';
+import {PanelTable} from 'sentry/components/panels';
+import {t} from 'sentry/locale';
+import EventView from 'sentry/utils/discover/eventView';
+import type {Sort} from 'sentry/utils/discover/fields';
+import getRouteStringFromRoutes from 'sentry/utils/getRouteStringFromRoutes';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOrganization from 'sentry/utils/useOrganization';
+import {useRoutes} from 'sentry/utils/useRoutes';
+import type {ReplayListRecordWithTx} from 'sentry/views/performance/transactionSummary/transactionReplays/useReplaysFromTransaction';
+import HeaderCell from 'sentry/views/replays/replayTable/headerCell';
+import TableCell from 'sentry/views/replays/replayTable/tableCell';
+import type {VisibleColumns} from 'sentry/views/replays/replayTable/types';
+import type {ReplayListRecord} from 'sentry/views/replays/types';
+
+type Props = {
+  fetchError: undefined | Error;
+  isFetching: boolean;
+  replays: undefined | ReplayListRecord[] | ReplayListRecordWithTx[];
+  sort: Sort | undefined;
+  visibleColumns: VisibleColumns;
+};
+
+function ReplayTable({fetchError, isFetching, replays, sort, visibleColumns}: Props) {
+  const routes = useRoutes();
+  const location = useLocation();
+  const referrer = getRouteStringFromRoutes(routes);
+
+  const organization = useOrganization();
+
+  const tableHeaders = [
+    visibleColumns.session ? <HeaderCell column="session" sort={sort} /> : null,
+    visibleColumns.projectId ? <HeaderCell column="projectId" sort={sort} /> : null,
+    visibleColumns.slowestTransaction ? (
+      <HeaderCell column="slowestTransaction" sort={sort} />
+    ) : null,
+    visibleColumns.startedAt ? <HeaderCell column="startedAt" sort={sort} /> : null,
+    visibleColumns.duration ? <HeaderCell column="duration" sort={sort} /> : null,
+    visibleColumns.countErrors ? <HeaderCell column="countErrors" sort={sort} /> : null,
+    visibleColumns.activity ? <HeaderCell column="activity" sort={sort} /> : null,
+  ].filter(Boolean);
```

Comment:
This assumes HeaderCell.column == keyof visibleColumns...

```suggestion
  const tableHeaders = Object.entries(visibleColumns).filter(([, visible]) => visible).map(([column]) => <HeaderCell column={column} sort={sort});
```

Reply:
I'm worried about the display order of this.... object keys are not technically sorted. 

maybe I can switch the visibleColumns object into an array though.

Reply:
switched!

Reply:
I was curious if this was still the case re keys in insertion order and it looks like in es2015, only certain methods (`keys()` being excluded) would guarantee order for non-int keys. But as of es2020, all of them should guarantee order.